wallet/src/actors/worker/methods.rs

```diff
@@ -1013,7 +1015,13 @@ impl Worker {
         wallet: types::SessionWallet,
         sink: types::DynamicSink,
     ) -> Result<()> {
-        log::debug!("The node has changed its status into {:?}", status);
+        log::info!("The node has changed its status into {:?}", status);
+        if status == StateMachine::Synced && !wallet.is_syncing()? {
+            wallet.clear_pending_data().ok();
```

Comment:
This will delete any pending transactions? What happens if we call `self.sync` without clearing the pending data? It should work, right?

Reply:
Not sure, the issue #1704 specifies to clear the pending data. Anyway it depends on the data, I think in this case we are considering data that is not included in the superblock, but maybe we should make sure if the data is actually pending or if it was included in a superblock while the node was nos synced.